### ISSUE TITLE
SQL Errors (type "geometry" or function/relation does not exist) when executing queries using PostGIS features (e.g. casting to geometry type)

### ISSUE THREAD

**Issue by Chealer** [dbeaver/dbeaver#7756]
#### System information: 
- Windows 8
- DBeaver 6.3.3 to 6.3.5
- No additional extensions

#### Connection specification:
- PostgreSQL 10.10 with PostGIS 2.4.3
- PostgreSQL JDBC Driver
- No tunnels or proxies

#### Describe the problem you're observing:
Trying to execute a query which casts to the geometry type fails. This type is defined by PostgreSQL's PostGIS extension. The same query works fine with psql or pgAdmin.

#### Steps to reproduce, if exist:
1. Open a console on a PostgreSQL connection with PostGIS
2. Enter a query like the following:
```
SELECT 1 WHERE ST_Intersects(
ST_MULTI(
ST_GeomFromText('POLYGON((0 0, 0 1, 1 1, 1 0, 0 0))',4326)
)::geometry, 
ST_MULTI(
ST_GeomFromText('POLYGON((0 0, 0 1, 1 1, 1 0, 0 0))',4326)
)::geometry
)
```
or just
`SELECT ST_GeomFromText('POLYGON((0 0, 0 1, 1 1, 1 0, 0 0))',4326)::geometry`
3. Execute the statement

#### Include any warning/errors/backtraces from the logs
The Error Log shows the following:
> SQL Error [42704]: ERROR: type "geometry" does not exist

dbeaver-debug.log contains:

> 2020-01-28 11:38:44.280 - Create display
> > Start Eclipse IDE UI Application [org.eclipse.ui.ide.application 1.3.500.v20191122-2109]
> 2020-01-28 11:38:44.371 - DBeaver 6.3.3.202001191557 is starting
> 2020-01-28 11:38:44.373 - OS: Windows 8.1 6.3 (amd64)
> 2020-01-28 11:38:44.373 - Java version: 11.0.5 by AdoptOpenJDK (64bit)
> 2020-01-28 11:38:44.373 - Install path: 'file:/C:/Program Files/DBeaver/'
> 2020-01-28 11:38:44.373 - Instance path: 'file:/C:/Users/CLOP03/AppData/Roaming/DBeaverData/workspace6/'
> 2020-01-28 11:38:44.373 - Memory available 64Mb/1024Mb
> > Start Core File Systems [org.eclipse.core.filesystem 1.7.600.v20191122-2104]
> > Start Core Resource Management [org.eclipse.core.resources 3.13.600.v20191122-2104]
> 2020-01-28 11:38:44.526 - Starting RMI server at 47425
> 2020-01-28 11:38:44.588 - Run workbench
> > Start Eclipse e4 Workbench SWT [org.eclipse.e4.ui.workbench.swt 0.14.800.v20190930-1643]
> > Start Eclipse Application Services [org.eclipse.e4.core.services 2.2.100.v20191122-2104]
> > Start Eclipse Contexts [org.eclipse.e4.core.contexts 1.8.300.v20191017-1404]
> > Start Eclipse UI Application Services [org.eclipse.e4.ui.services 1.3.700.v20190930-1643]
> > Start Eclipse UI Dependency Injection [org.eclipse.e4.ui.di 1.2.700.v20191114-0830]
> > Start ICU4J Base [com.ibm.icu.base 58.2.0.v20170418-1837]
> > Start EMF Common [org.eclipse.emf.common 2.17.0.v20190920-0401]
> > Start EMF Ecore [org.eclipse.emf.ecore 2.20.0.v20190920-0401]
> > Start EMF XML/XMI Persistence [org.eclipse.emf.ecore.xmi 2.16.0.v20190528-0725]
> > Start Eclipse e4 Workbench Add-ons [org.eclipse.e4.ui.workbench.addons.swt 1.3.700.v20191030-1314]
> > Start Expression Language [org.eclipse.core.expressions 3.6.600.v20191122-2104]
> > Start Eclipse e4 Workbench SWT Renderer [org.eclipse.e4.ui.workbench.renderers.swt 0.14.900.v20191114-0715]
> > Start Eclipse Bindings Support [org.eclipse.e4.ui.bindings 0.12.700.v20191105-1310]
> > Start Eclipse Dependency Injection Extensions [org.eclipse.e4.core.di.extensions 0.15.400.v20191122-2104]
> > Start Eclipse CSS SWT Support [org.eclipse.e4.ui.css.swt 0.13.700.v20191113-1031]
> > Start Eclipse CSS SWT Theme Support [org.eclipse.e4.ui.css.swt.theme 0.12.500.v20191125-1011]
> > Start DBeaver UI [org.jkiss.dbeaver.ui 5.1.40.202001191557]
> 2020-01-28 11:38:45.971 - Initializing DBeaver 6.3.3
> 2020-01-28 11:38:45.972 - Host plugin: org.jkiss.dbeaver.core.application 6.3.3.202001191557
> 2020-01-28 11:38:45.972 - Initialize Core...
> 2020-01-28 11:38:45.973 - Can't find BC security provider. Use default JCE.
> 2020-01-28 11:38:45.973 - BouncyCastle not registered, using the default JCE provider
> 2020-01-28 11:38:46.068 - Initialize base platform...
> > Start DBeaver UI Navigator [org.jkiss.dbeaver.ui.navigator 1.0.25.202001191557]
> > Start Equinox Java Authentication and Authorization Service (JAAS) [org.eclipse.equinox.security 1.3.400.v20191008-0645]
> > Start Internet Connection Management UI [org.eclipse.ui.net 1.3.600.v20190925-1021]
> > Start Internet Connection Management [org.eclipse.core.net 1.3.700.v20191122-2107]
> 2020-01-28 11:38:46.108 - Proxy service 'org.eclipse.core.net.proxy.IProxyService' loaded
> 2020-01-28 11:38:46.111 - Core initialized (139ms)
> 2020-01-28 11:38:46.111 - Configure workbench window
> > Start Help System Core [org.eclipse.help 3.8.600.v20191123-0656]
> 2020-01-28 11:38:46.721 - Create actions
> 2020-01-28 11:38:46.723 - Disable Eclipse action set 'org.eclipse.ui.edit.text.actionSet.annotationNavigation'
> 2020-01-28 11:38:46.723 - Disable Eclipse action set 'org.eclipse.ui.WorkingSetActionSet'
> > Start Default Text Editor [org.eclipse.ui.editors 3.13.0.v20191203-1122]
> > Start File Buffers [org.eclipse.core.filebuffers 3.6.800.v20191122-2108]
> > Start Text Editor Framework [org.eclipse.ui.workbench.texteditor 3.14.0.v20191122-2108]
> 2020-01-28 11:38:47.454 - Initialize workbench window
> 2020-01-28 11:38:47.462 - Finish initialization
> > Start DBeaver UI Editors - Base [org.jkiss.dbeaver.ui.editors.base 1.0.25.202001191557]
> > Start DBeaver SQL Model [org.jkiss.dbeaver.model.sql 1.0.15.202001191557]
> > Start DBeaver UI Editors - SQL [org.jkiss.dbeaver.ui.editors.sql 1.0.25.202001191557]
> > Start Team Support Core [org.eclipse.team.core 3.8.800.v20191122-2107]
> > Start Core Compare Support [org.eclipse.compare.core 3.6.700.v20191122-2107]
> > Start Log View [org.eclipse.ui.views.log 1.2.900.v20191122-2109]
> > Start Equinox Provisioning Core [org.eclipse.equinox.p2.core 2.6.200.v20191014-1220]
> > Start Equinox Provisioning Operations API [org.eclipse.equinox.p2.operations 2.5.600.v20191016-0507]
> > Start Equinox Provisioning Platform Automatic Update Support [org.eclipse.equinox.p2.ui.sdk.scheduler 1.4.500.v20191016-0512]
> > Start Equinox Provisioning Metadata [org.eclipse.equinox.p2.metadata 2.4.600.v20191002-0702]
> > Start Equinox Provisioning Update Checker [org.eclipse.equinox.p2.updatechecker 1.2.200.v20190701-1309]
> > Start Equinox Provisioning Engine [org.eclipse.equinox.p2.engine 2.6.500.v20191014-1645]
> > Start Equinox Provisioning Metadata Repository [org.eclipse.equinox.p2.metadata.repository 1.3.300.v20191002-0702]
> > Start Equinox Provisioning Repository [org.eclipse.equinox.p2.repository 2.4.600.v20191016-0510]
> > Start Equinox Provisioning UI Support [org.eclipse.equinox.p2.ui 2.5.700.v20190930-1517]
> 2020-01-28 11:38:49.322 - Total database drivers: 81 (81)
> > Start DBeaver UI Editors - Data [org.jkiss.dbeaver.ui.editors.data 1.0.25.202001191557]
> > Start DBeaver Sample Database [org.jkiss.dbeaver.ext.sample.database 1.0.62.202001191557]
> > Start DBeaver Tip of the day [org.jkiss.dbeaver.ext.ui.tipoftheday 1.0.31.202001191557]
> > Start DBeaver ER Diagram [org.jkiss.dbeaver.ext.erd 2.0.127.202001191557]
> > Start Console [org.eclipse.ui.console 3.9.0.v20191126-0912]
> 2020-01-28 11:38:49.939 - Connect with 'GO DEV admgeo1' (postgres-jdbc-16fe8ffb2af-3da7c7a2aa87f2c7)
> > Start DBeaver PostgreSQL Support [org.jkiss.dbeaver.ext.postgresql 2.1.98.202001191557]
> 2020-01-28 11:38:50.015 - Initiate connection to PostgreSQL database [pgsyst@spssogl97d] for Main <pgsyst>
> 2020-01-28 11:38:50.282 - Initiate connection to PostgreSQL database [pgsyst@spssogl97d] for Metadata <pgsyst>
> 2020-01-28 11:38:50.400 - Connected (postgres-jdbc-16fe8ffb2af-3da7c7a2aa87f2c7, PostgreSQL JDBC Driver [42.2.5])
> 2020-01-28 11:38:50.431 - Initiate connection to PostgreSQL database [pgsyst@spssogl97d] for SQLEditor <Script.sql>
> 2020-01-28 11:38:52.024 - Open [https://dbeaver.io/product/version.xml]
> 2020-01-28 11:39:08.776 - SQL Error [42704]: ERROR: type "geometry" does not exist
>   Position: 106
> org.jkiss.dbeaver.model.sql.DBSQLException: SQL Error [42704]: ERROR: type "geometry" does not exist
>   Position: 106
> 	at org.jkiss.dbeaver.model.impl.jdbc.exec.JDBCStatementImpl.executeStatement(JDBCStatementImpl.java:134)
> 	at org.jkiss.dbeaver.ui.editors.sql.execute.SQLQueryJob.executeStatement(SQLQueryJob.java:486)
> 	at org.jkiss.dbeaver.ui.editors.sql.execute.SQLQueryJob.lambda$0(SQLQueryJob.java:424)
> 	at org.jkiss.dbeaver.model.exec.DBExecUtils.tryExecuteRecover(DBExecUtils.java:159)
> 	at org.jkiss.dbeaver.ui.editors.sql.execute.SQLQueryJob.executeSingleQuery(SQLQueryJob.java:416)
> 	at org.jkiss.dbeaver.ui.editors.sql.execute.SQLQueryJob.extractData(SQLQueryJob.java:776)
> 	at org.jkiss.dbeaver.ui.editors.sql.SQLEditor$QueryResultsContainer.readData(SQLEditor.java:2861)
> 	at org.jkiss.dbeaver.ui.controls.resultset.ResultSetJobDataRead.lambda$0(ResultSetJobDataRead.java:110)
> 	at org.jkiss.dbeaver.model.exec.DBExecUtils.tryExecuteRecover(DBExecUtils.java:159)
> 	at org.jkiss.dbeaver.ui.controls.resultset.ResultSetJobDataRead.run(ResultSetJobDataRead.java:108)
> 	at org.jkiss.dbeaver.ui.controls.resultset.ResultSetViewer$17.run(ResultSetViewer.java:3454)
> 	at org.jkiss.dbeaver.model.runtime.AbstractJob.run(AbstractJob.java:103)
> 	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)
> Caused by: org.postgresql.util.PSQLException: ERROR: type "geometry" does not exist
>   Position: 106
> 	at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2440)
> 	at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2183)
> 	at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:308)
> 	at org.postgresql.jdbc.PgStatement.executeInternal(PgStatement.java:441)
> 	at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:365)
> 	at org.postgresql.jdbc.PgStatement.executeWithFlags(PgStatement.java:307)
> 	at org.postgresql.jdbc.PgStatement.executeCachedSql(PgStatement.java:293)
> 	at org.postgresql.jdbc.PgStatement.executeWithFlags(PgStatement.java:270)
> 	at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:266)
> 	at org.jkiss.dbeaver.model.impl.jdbc.exec.JDBCStatementImpl.execute(JDBCStatementImpl.java:338)
> 	at org.jkiss.dbeaver.model.impl.jdbc.exec.JDBCStatementImpl.executeStatement(JDBCStatementImpl.java:131)
> 	... 12 more

**Comment by kseniiaguzeeva** [comment]
Unfortunately I cannot reproduce it. Could you please make a screenshot/video of your steps to help to recognize the issue?

**Comment by Chealer** [comment]
@kseniiaguzeeva I did (but this tracker does not hosting it).

**Comment by kseniiaguzeeva** [comment]
Could you recheck whether a correct schema is selected (use horkey "Ctrl+0" in SQL Editor)

**Comment by Chealer** [comment]
> Could you recheck whether a correct schema is selected (use horkey "Ctrl+0" in SQL Editor)

Yes, this reproduces even if a schema is selected.

**Comment by serge-rider** [comment]
@Chealer and are you sure that PostGIS is installed in this schema?
You can see it in Extension folder under your database

**Comment by Chealer** [comment]
> @Chealer and are you sure that PostGIS is installed in this schema?
> You can see it in Extension folder under your database

Yes, but thank you for the tip. By the way, I got the following error trying to open the Extensions folder:

> ERROR: permission denied for relation pg_authid

Should that be tracked in a different ticket?

**Comment by Chealer** [comment]
I am generalizing the ticket's title since this also happens with very different queries, like the following:
```
insert into msp_limite_quebec_s
select ST_GeometryN(geom_s,1)::Geometry(Polygon, 4326), -1200 from (
select (st_dump(st_transform(st_simplify(st_buffer(st_union(geom_s), -1200),350),4326))).geom as geom_s
from synonyms.msp_adm_reg_admin_s
) a
limit 1;
```

> SQL Error [42883]: ERROR: function st_union(public.geometry) does not exist
>   Hint: No function matches the given name and argument types. You might need to add explicit type casts.
>   Position: 155

**Comment by Chealer** [comment]
Here is another problem which I assume has the same cause.
Query:
`refresh materialized view msp_masas_base_entry_vigilance_v_p;`
Error:

> SQL Error [42P01]: ERROR: relation "spatial_ref_sys" does not exist

**Comment by uslss** [comment]
Cannot reproduce issue against 7.2.0.
If issue is still actual - feel free to reopen an issue

**Comment by Chealer** [comment]
This does seem to be solved in 7.1.5.
